src/GraphQL.Client/GraphQLHttpClientOptions.cs

```diff
@@ -56,5 +56,11 @@ public class GraphQLHttpClientOptions
         /// Configure additional websocket options (i.e. headers). This will not be invoked on Windows 7 when targeting .NET Framework 4.x. 
         /// </summary>
         public Action<ClientWebSocketOptions> ConfigureWebsocketOptions { get; set; } = options => { };
+
+        /// <summary>
+        /// The `ConnectionParams` object sent with the GQL_CONNECTION_INIT message on establishing a websocket connection.
+        /// See https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md#gql_connection_init.
+        /// </summary>
+        public object? WebSocketConnectionParams { get; set; } = null;
```

Comment:
I advise you to change the type of this property to delegate. Often GQL_CONNECTION_INIT message is used to convey authentication information. This data is different for different users. Note that the properties above have already been made by delegates for the same reason - this is more flexible.

Reply:
Also consider to converge naming. I can see `WebSocketConnectionParams`, `ConnectionParams`, `Payload` terms now both describing the same thing. I would prefer to use `Payload` term.

Reply:
I took the name `ConnectionParams` from the Apollo spec, to stay close to the naming used there... most "tutorials" are for Appollo anyway...

I think its confusing if we call this property on the options class just `Payload`...

Reply:
I renamed it to `SetWebSocketConnectionInitPayload`, I suppose that's what you meant in the first place...